backend/src/services/auctions.ts

```diff
@@ -69,8 +72,50 @@ class AuctionsService {
       buyerId: auction.buyerId?.toString(),
       expectedValue: auction.expectedValue,
       createdDate: auction.createdAt,
+      status: auction.status,
+      finalBidAmount: auction.finalBidAmount,
     };
   }
+  // Close an auction, determining the winner if there are any bids.
+  static async closeAuction(auctionId: string): Promise<void> {
+    const auction = await Auction.findById(auctionId);
+    if (!auction) throw new Error('auction not found');
+
+    // Already closed
+    if (auction.status !== 'active') return;
+
+    // Fetch bids 
+    const bids = await Bid.find({ auctionId }).sort({ amount: -1 });
+
+    // No bids 
+    if (bids.length === 0) {
+      auction.status = 'closed';
+      await auction.save();
+      await Auction.findByIdAndDelete(auctionId);
+      return;
+    }
+
+    // Highest bid
+    const highest = bids[0];
+
+    auction.buyerId = highest.userId;      
+    auction.finalBidAmount = highest.amount;
+    auction.status = 'closed';
```

Comment:
Should this not be `purchased`? Just going by what you had in the DB model.


```suggestion
    auction.status = 'purchased';
```

Reply:
Corrected.